### ISSUE TITLE
ExpiredTokenException: The security token included in the request is expired

### ISSUE THREAD

**Issue by kldeb** [sst/ion#231]
Steps to reproduce:

1. Create a remix app
2. Create a postress db
3. Link postgres to remix
4. npm install drizzle, @aws-sdk/client-rds-data
5. export const db = drizzle(new RDSDataClient({}), {
   database: Resource.MyDatabase.database,
   secretArn: Resource.MyDatabase.secretArn,
   resourceArn: Resource.MyDatabase.clusterArn,
   });
6. run `npm run dev`
7. Let the localhost app sit idle for \~20 mins
8. see the following error:

```
ExpiredTokenException: The security token included in the request is expired
    at throwDefaultError (/Users/kldeb/dev/lop-ion/node_modules/@smithy/smithy-client/dist-cjs/index.js:838:20)
    at /Users/kldeb/dev/lop-ion/node_modules/@smithy/smithy-client/dist-cjs/index.js:847:5
    at de_CommandError (/Users/kldeb/dev/lop-ion/node_modules/@aws-sdk/client-rds-data/dist-cjs/index.js:830:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/kldeb/dev/lop-ion/node_modules/@smithy/middleware-serde/dist-cjs/index.js:35:20
    at async /Users/kldeb/dev/lop-ion/node_modules/@smithy/core/dist-cjs/index.js:165:18
    at async /Users/kldeb/dev/lop-ion/node_modules/@smithy/middleware-retry/dist-cjs/index.js:320:38
    at async /Users/kldeb/dev/lop-ion/node_modules/@aws-sdk/middleware-logger/dist-cjs/index.js:33:22
    at async AwsDataApiPreparedQuery.values (file:///Users/kldeb/dev/lop-ion/node_modules/drizzle-orm/aws-data-api/pg/session.js:65:20)
    at async AwsDataApiPreparedQuery.execute (file:///Users/kldeb/dev/lop-ion/node_modules/drizzle-orm/aws-data-api/pg/session.js:41:18)
    at async loader (/Users/kldeb/dev/lop-ion/app/routes/_app.leads.$id.step-4.tsx:56:29)
    at async Object.callRouteLoaderRR (/Users/kldeb/dev/lop-ion/node_modules/@remix-run/server-runtime/dist/data.js:52:16)
    at async callLoaderOrAction (/Users/kldeb/dev/lop-ion/node_modules/@remix-run/router/dist/router.cjs.js:4166:16)
    at async Promise.all (index 1)
    at async loadRouteData (/Users/kldeb/dev/lop-ion/node_modules/@remix-run/router/dist/router.cjs.js:3638:19)
    at async queryImpl (/Users/kldeb/dev/lop-ion/node_modules/@remix-run/router/dist/router.cjs.js:3481:20)
    at async Object.query (/Users/kldeb/dev/lop-ion/node_modules/@remix-run/router/dist/router.cjs.js:3381:18)
    at async handleDocumentRequestRR (/Users/kldeb/dev/lop-ion/node_modules/@remix-run/server-runtime/dist/server.js:184:15)
    at async requestHandler (/Users/kldeb/dev/lop-ion/node_modules/@remix-run/server-runtime/dist/server.js:108:18)
    at async nodeHandler (/Users/kldeb/dev/lop-ion/node_modules/@remix-run/dev/dist/vite/plugin.js:837:27)
    at async /Users/kldeb/dev/lop-ion/node_modules/@remix-run/dev/dist/vite/plugin.js:840:15```
```

**Comment by thdxr** [comment]
does it work when restarted?

**Comment by kldeb** [comment]
Yes

**Comment by magicspon** [comment]
Do I need to worry about this in production?

**Comment by jayair** [comment]
This is related to `sst dev` I think. So it shouldn't be production related.

**Comment by jayair** [comment]
Yeah we should take a look at this.

**Comment by kldeb** [comment]
Also have this problem with the dynamo db client. Doesn't happen when using SSO.

**Comment by ArnaudMiguet** [comment]
Hi ! I’m seeing the same kind of error with S3Client (with or without presigned url). I create it in a server action in next to upload to s3, and it works at first and then after some time (< 1h), it returns TokenExpired errors. I have sst up to date (3.0.75). Any news on this ?

**Comment by jayair** [comment]
@ArnaudMiguet what kind of credentials are you using locally? Are they expiring by any chance?

**Comment by ArnaudMiguet** [comment]
I'm using the access key for an IAM user with AdministratorAccess role. Not sure if and why the token used by the s3client would expire.

**Comment by ArnaudMiguet** [comment]
Another piece of information : it does look like it is local environment related, it happens on both my machines, one is a Mac and the other is a pc running sst dev through WSL.

**Comment by ArnaudMiguet** [comment]
I was using my root account with the aws/credentials file and it seems like the session token expires after one hour.
I now tried to setup the aws iam identity center with prod and dev accounts following the sst guide. I used an "AdministratorAccess" role with 1h expiry just to try, and it now seems like the session token gets renewed because I've been able to work on my project for more than one hour. So it is working now. I just lost all my test databases and buckets because of the new account, but not a problem :) Could it be related to the fact that i used the credentials for the access to my root account directly ?
